src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -22,12 +32,16 @@
           (log/warn (format "%s %s"
                             (i18n/trs "The ''master.allow-header-cert-info'' setting is deprecated and will be ignored in favor of the ''authorization.allow-header-cert-info'' setting because the ''jruby-puppet.use-legacy-auth-conf'' setting is ''false''.")
                             (i18n/trs "Remove the ''master.allow-header-cert-info'' setting.")))))
-      (assoc context :request-handler
-                     (request-handler-core/build-request-handler
-                      jruby-service
-                      (request-handler-core/config->request-handler-settings
-                       config)
-                      current-code-id))))
+      ;; TODO: Throw error when max-active-requests > 0, but metrics-service is nil
```

Comment:
> The problem here is that prioritization alone doesn't solve the issue of the thundering herd.  In order to break a herd up, we need to tell agents to apply a randomized backoff so that their runtimes de-sync.

Sure, I wasn't questioning that aspect of this.  I think you're right about that.  Was just thinking if we had the ability to suspend requests with prioritization that we could prioritize ones like `/environment_classes` and `/status` ahead of ones that are typically agent-driven like `../catalog` - hopefully making it a lot less likely that we'd ever need to drop the higher priority ones.  If we end up pursuing a QoSFilter approach for prioritization later, I think it would be good to consider how to factor in a randomized splay for requests rejected with a 503 as your PR does.